File: app/src/main/java/com/neome/java/api/meta/base/dto/DefnFieldFormList.java
// neome.ai API - do not change
//

package com.neome.java.api.meta.base.dto;

import com.neome.java.api.meta.base.Types.MetaIdField;
import com.neome.java.api.meta.base.Types.MetaIdSection;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class DefnFieldFormList extends DefnField
{
  @Nullable
  public MetaIdField bgColorFieldId;

  @Nullable
  public Boolean cardLayoutItemHideBorders;

  @Nullable
  public Long cardLayoutNumOfColumns;

  @Nullable
  public MetaIdField displayItemId;

  @Nullable
  public MetaIdSection displaySectionId;

  @Nullable
  public MetaIdField[] editableFieldIdSet;

  @Nullable
  public Boolean hideMenu;

  @Nullable
  public Boolean ignoreSelection;

  @Nullable
  public Boolean isPickMany;

  @Nullable
  public Long itemHeight;

  @Nullable
  public DefnDtoLayoutCardItem layout;

  @Nullable
  public MetaIdField[] pickRowOnFieldIdSet;

  @Nullable
  public Boolean showAsCardLayout;
}

File: app/src/main/java/com/neome/api/meta/base/dto/DefnFieldFormList.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdSection

interface DefnFieldFormList : DefnField {
    val bgColorFieldId: MetaIdField?
    val cardLayoutItemHideBorders: Boolean?
    val cardLayoutNumOfColumns: Long?
    val displayItemId: MetaIdField?
    val displaySectionId: MetaIdSection?
    val editableFieldIdSet: List<MetaIdField>?
    val hideMenu: Boolean?
    val ignoreSelection: Boolean?
    val isPickMany: Boolean?
    val itemHeight: Long?
    val layout: DefnDtoLayoutCardItem?
    val pickRowOnFieldIdSet: List<MetaIdField>?
    val showAsCardLayout: Boolean?
}
